tests/bsd-user/hello-main/meson.build

```diff
@@ -0,0 +1,18 @@
+archs = ['armv7', 'amd64', 'aarch64', 'i386', 'mips',
+        'mipsel', 'mips64', 'powerpc', 'powerpc64', 'riscv64']
+
+foreach a : archs
+    custom_target('h.' + a,
+        command: [meson.get_compiler('c').cmd_array()[0], '-target', a + '-unknown-freebsd14.0', '-nostdlib', '-Wl,-e', '-Wl,qemu_start', '-static', '-o', '@OUTPUT@', '@INPUT@'],
```

Comment:
I think this bakes in that we have clang in the host environment.

Reply:
This is the compiler that is used ti build `qemu` itself. Is this wrong?

Reply:
Ah, you want it to be buildable with `gcc` too? I guess, I can handle that too.

Reply:
I'm thinking we might want to just grab the binaries from somewhere. We'll need to do that when we switch to testing using rescue anyway...

Reply:
But this way we can quickly add more tests that checks some specific instructions, for instance. I think, we need to have both ways of getting binaries.

Reply:
If we want to do that, we should leverage the native qemu stuff they have for testing that. I was sent patches recently to get it to compile on BSD.

Reply:
Ok, then what's the path forward now? Abandon this and look into how qemu does testing natively?

Before doing this PR I looked up various directories inside `tests/` but didn't find anything close to this.

Reply:
sorry to let this linger for a few days... Life, eh?

The path forward is to just generate the binaries and check those into the test. That way we don't have the build wrinkle to deal with and we get the testing we need.

Reply:
Longer term we need something else, but for now this would be good. The something else would be external to qemu and would likely be an A/B comparison of 'old' and 'test' qemu-bsd-user binaries for something like a kyua run.

Reply:
Where to fetch the binaries from, then?